app/controllers/gift_requests_controller.rb

```diff
@@ -40,14 +41,31 @@ def destroy_multiple
     end
   end
 
+  def catalog
+    @back_route = events_path
+
+    @event = Event.find(params[:id])
+    @gift_requests = @event.gift_requests.unreserved.order(created_at: :asc).page(params[:page] || 1)
+  end
+
   private
 
   def gift_request_params
     params.require(:gift_request).permit(:id, :name, :recipient, :description, :link, :gender, :age)
   end
 
+  def require_therapist
+    unless current_user.try(:therapist?)
+      redirect_to root_path, alert: "You are not authorized to access that page."
```

Comment:
I would avoid that for two reasons:

1. The user should never be able to use a link on the site to get to a section they aren't allowed to access, so these pages are accessed independent of normal user flow.
2. `redirect_to :back` is unreliable and will blow up if there's no 'back' defined, so I prefer to avoid it where possible.

In the spirit of JSON APIs, though, I think when you hit an 'unauthorized' page it should give you a way to become authorized, so I'll switch this up to take you to the login page instead if the user isn't logged in.